pkg/translator/api/util.go

```diff
@@ -22,10 +25,11 @@ const (
 	edgeLBPoolNameSuffixLength = 5
 )
 
-// NewRandomEdgeLBPoolName returns a string meant to be used as the name of an EdgeLB pool.
+// newRandomEdgeLBPoolName returns a string meant to be used as the name of an EdgeLB pool.
 // The computed name is of the form "[<prefix>--]<cluster-name>--<suffix>", where "<prefix>" is the specified (possibly empty) string and "<suffix>" is a randomly-generated suffix.
 // It is guaranteed not to exceed 63 characters.
-func NewRandomEdgeLBPoolName(prefix string) string {
+// It is also guaranteed, to the best of our ability, not to clash with the names of any pre-existing EdgeLB pool.
+func newRandomEdgeLBPoolName(prefix string) string {
```

Comment:
Why not use a string pointer instead, since an empty string is a valid input?

Reply:
For basic data types, I _usually_ use pointers only when I need values to be modified from inside a `func`  (exceptions apply, as always), or when I absolutely need to check for `nil` values (which is not the case here). Plus, I think that in this case that would over-complicate the body of the `func` for no added benefit.